sqlalchemy_redshift/dialect.py

```diff
@@ -547,12 +543,11 @@ def create_connect_args(self, *args, **kwargs):
         return cargs, default_args
 
     def _get_table_or_view_names(self, relkind, connection, schema=None, **kw):
-        default_schema = inspect(connection).default_schema_name
         if not schema:
-            schema = default_schema
+            schema = self.default_schema_name
```

Comment:
Yes. See here: http://docs.sqlalchemy.org/en/latest/core/internals.html#sqlalchemy.engine.interfaces.Dialect. And the psycopg2 dialect which this dialect inherits from of course ultimately inherits from Dialect and implements this property.